.github/workflows/ci.yml

```diff
@@ -101,35 +104,36 @@ jobs:
       - name: Run linters
         run: poetry run invoke lint
 
-
-  deploy:
-    name: Deploy
+  prep-release:
+    name: Release preparations
     environment: Deployment
     needs: [test, lint]
     runs-on: ubuntu-latest
     if: ${{ github.ref=='refs/heads/main' && github.event_name!='pull_request' }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: "3.8"
       - name: Check release
         id: check_release
         run: |
           python -m pip install --upgrade pip
-          python -m pip install poetry githubrelease httpx==0.16.1 autopub
-          echo "##[set-output name=release;]$(autopub check)"
-      - name: Publish
+          python -m pip install poetry autopub[github]
+          echo "release=$(autopub check)" >> "$GITHUB_OUTPUT"
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "release<<$EOF" >> "$GITHUB_OUTPUT"
+          autopub check >> "$GITHUB_OUTPUT"
+          echo "$EOF" >> "$GITHUB_OUTPUT"
+      - name: Tag & Create GH release
         if: ${{ steps.check_release.outputs.release=='' }}
         env:
           GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
-          PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
         run: |
           git remote set-url origin https://$GITHUB_TOKEN@github.com/${{ github.repository }}
           autopub prepare
-          poetry build
+          autopub build
           autopub commit
           autopub githubrelease
```

Comment:
I don't like that the githubrelease upload has the packages from `autopub build` and the pypi upload builds it's own packages. I'd also love to use `build-and-inspect-python-package` from the `pypi-package` workflow to be able to inspect the outputs properly before. I guess I might move `autopub githubrelease` into the other workflow and run the other workflow on tags only? This would still result in a tag being pushed but we could then only run `autopub githubrelease` & pypi upload after we checked the output from `build-and-inspect-python-package` manually?

Reply:
Agreed — this seems like a more sensible approach. 👍

Reply:
Arg, the tags are not created by `autopub commit` but by `autopub githubrelease`. Honestly autopub is really trying to make this hard here. But I am somewhat out of ideas :/

Reply:
The tag is created as part of the `githubrelease` step because the latter needs to know the tag in order to create the release. It might be feasible to separate the tagging step into a separate command (`autopub tag`), but then we would need some way to maintain "state" such that the tag info could be passed to the `githubrelease` step.

Reply:
Mhm the state could be the tag (?) of the current commit?

Reply:
In an effort to find a more efficient place for us to discuss this topic, I created an AutoPub issue along with a proposal. What do you think? https://github.com/autopub/autopub/issues/39